ast/parser_test.go

```diff
@@ -2150,7 +2150,9 @@ func TestRuleRefHeads(t *testing.T) {
 		})
 	}
 
-	// TODO(sr): error cases, non-ground terms anywhere but at the end of the ref
```

Comment:
I've removed the TODO. There may be more error cases to add here, but that's implicit.

Also the "only string terms anywhere but at the end" part has been covered where it's enforced: in a compiler stage.